File: scripts/sce.js
#!/root/.nvm/versions/node/v7.3.0/bin/node

var expr1 = function () {
  return "return value of expr1.";
}

var expr2 = function () {
  console.log("Not Short-circuit evaluation.");
  return "return value of expr2.";
}

console.log(expr1() || expr2());

File: scripts/sce.f
      PROGRAM hello
          logical expr1, expr2
          WRITE (*,'(L)') (expr1().or.expr2())
      END

      logical FUNCTION expr1() result(r)
          r = .TRUE.
      END

      logical FUNCTION expr2() result(r)
          WRITE (*,'(A)') 'Not Short-circuit evaluation.'
          r = .TRUE.
      END
